signer/src/block_observer.rs

```diff
@@ -266,37 +280,31 @@ where
         self.write_stacks_blocks(&stacks_blocks).await?;
         self.write_bitcoin_block(&block).await?;
 
-        self.extract_deposit_requests(&block.txdata, block.block_hash())
-            .await?;
-
         tracing::debug!("finished processing bitcoin block");
         Ok(())
     }
 
-    async fn extract_deposit_requests(
-        &mut self,
-        txs: &[Transaction],
-        block_hash: BlockHash,
-    ) -> Result<(), Error> {
-        let (deposit_request, deposit_request_txs) = txs
-            .iter()
-            .filter_map(|tx| self.deposit_requests.remove(&tx.compute_txid()))
-            .flatten()
+    /// For each of the deposit requests, persist the corresponding
+    /// transaction and the parsed deposit info into the database.
+    ///
+    /// Since this functions writes to the `bitcoin_transactions` table, we
```

Comment:
```suggestion
    /// Since this function writes to the `bitcoin_transactions` table, we
```

Reply:
Done in https://github.com/stacks-network/sbtc/pull/679/commits/ef23cb0dc04ee25c8f76fc06cd7d6ac7acc3faa9